PWGJE/TableProducer/emcalCorrectionTask.cxx

```diff
@@ -237,173 +228,340 @@ struct EmcalCorrectionTask {
+      // Convert aod::Calo to o2::emcal::Cell which can be used with the clusterizer.
+      // In particular, we need to filter only EMCAL cells.
+
+      // Get the collisions matched to the BC using foundBCId of the collision
+      auto collisionsInBC = collisions.sliceBy(collisionsPerBC, bc.globalIndex());
+      auto cellsInBC = cells.sliceBy(cellsPerFoundBC, bc.globalIndex());
+
+      if (!cellsInBC.size()) {
+        LOG(debug) << "No cells found for BC";
+        countBC(bc.globalBC(), collisionsInBC.size(), false);
+        continue;
+      }
+      // Counters for BCs with matched collisions
+      countBC(bc.globalBC(), collisionsInBC.size(), true);
+      std::vector<o2::emcal::Cell> cellsBC;
+      std::vector<int64_t> cellIndicesBC;
+      for (auto& cell : cellsInBC) {
+        cellsBC.emplace_back(cell.cellNumber(),
+                             cell.amplitude(),
+                             cell.time(),
+                             o2::emcal::intToChannelType(cell.cellType()));
+        cellIndicesBC.emplace_back(cell.globalIndex());
+      }
+      LOG(detail) << "Number of cells for BC (CF): " << cellsBC.size();
+      nCellsProcessed += cellsBC.size();
+
+      fillQAHistogram(cellsBC, mHistManager);
+
+      // TODO: Helpful for now, but should be removed.
+      LOG(debug) << "Converted EMCAL cells";
+      for (auto& cell : cellsBC) {
+        LOG(debug) << cell.getTower() << ": E: " << cell.getEnergy() << ", time: " << cell.getTimeStamp() << ", type: " << cell.getType();
+      }
+
+      LOG(debug) << "Converted cells. Contains: " << cellsBC.size() << ". Originally " << cellsInBC.size() << ". About to run clusterizer.";
+
+      //  this is a test
+      //  Run the clusterizers
+      LOG(debug) << "Running clusterizers";
+      Int_t iClusterizer = 0;
+      for (auto& clusterizer : mClusterizers) {
+        clusterizer->findClusters(cellsBC);
+
+        auto emcalClusters = clusterizer->getFoundClusters();
+        auto emcalClustersInputIndices =
+          clusterizer->getFoundClustersInputIndices();
+        LOG(debug) << "Retrieved results. About to setup cluster factory.";
+
+        convertToAnalysisCluster(cellsBC, iClusterizer, emcalClusters, emcalClustersInputIndices);
+
+        float vx = 0, vy = 0, vz = 0;
+        if (collisionsInBC.size() == 1) {
+          // dummy loop to get the first collision
+          for (const auto& col : collisionsInBC) {
+            mHistManager.fill(HIST("hCollPerBC"), 1);
+            mHistManager.fill(HIST("hCollisionType"), 1);
+            vx = col.posX();
+            vy = col.posY();
+            vz = col.posZ();
+
+            std::vector<std::vector<int>> clusterToTrackIndexMap;
+            std::vector<std::vector<int>> trackToClusterIndexMap;
+            std::tuple<std::vector<std::vector<int>>, std::vector<std::vector<int>>> IndexMapPair = {clusterToTrackIndexMap, trackToClusterIndexMap};
+            std::vector<int64_t> trackGlobalIndex;
```

Comment:
Won't be needed in case the corresponding argument of `FillClusterTable` are std::optional